### DISCUSSION TITLE
How do I retrieve record fields from a collection?

### DISCUSSION THREAD

**Discussion by ItsTerm1n4l** [pocketbase/pocketbase#3931] [Q&A] [answered]
I am trying to use pocketbase as a database to store info on computer parts, having an individual record for each part and fields in that record containing info about that part. Like so.
![20231212_17h41m13s_grim](https://github.com/pocketbase/pocketbase/assets/118511626/c3aecf11-2ce3-40de-8f41-7b8307e66e77)
So I want to get the record and fields for various parts and display that information in my website.
I am using svelte as my frontend framework but I think everything should be that same as if I was using plain javascript.
Thank you.

**Comment by ganigeorgiev** [comment]
You can click on the "API Preview" and explore the snippets corresponding to the desired action OR check the generic web API documentation in https://pocketbase.io/docs/api-records/.

Note that by default only admins can access the collection records. To "unlock" it you'll have to configure your API rules according to your specific use case. More details about the API rules and its syntax you can find in https://pocketbase.io/docs/api-rules-and-filters/.

**Reply by ItsTerm1n4l** [reply]
I have tried the snippets for viewing records and have allowed access in the api rules but it still doesn't work, I just get a blank white screen when I try to access my website (my website does usually have visible content until I add the snippet), I think it might be something to do with the `await` in the `const await records...`.
Here is my basic code (no website design or comments).
```
<script>
  import svelteLogo from './assets/svelte.svg'
  import viteLogo from '/vite.svg'
  import Counter from './lib/Counter.svelte'

  import { Router, Link, Route } from "svelte-routing";
  import Home from "./pages/Home.svelte";
  import About from "./pages/About.svelte";
  import Site from "./routes/site.svelte";
  import Builder from "./pages/Builder.svelte";
  export let url = "";

import PocketBase from 'pocketbase';
import { writable } from 'svelte/store';

const pb = new PocketBase('http://127.0.0.1:8090'); 
const record = await pb.collection('parts').getOne('fg0c9n6avoq5efa', {
    expand: 'relField1,relField2.subRelField',
});
</script>
<p>{record}</p>
<style>

</style>
```

**Reply by ganigeorgiev** [reply, marked as answer]
I don't think this is PocketBase related. Make sure to check the browser console or the terminal for JS errors.
I haven't tested it but Svelte may not allow top-level `await` in the `script` blocks and note that reactive variables usually are defined with `let`. Try:

```js
const pb = new PocketBase('http://127.0.0.1:8090'); 

let record;

load();

async function load() {
    record = await pb.collection('parts').getOne('fg0c9n6avoq5efa');
}
```

**Reply by ItsTerm1n4l** [reply]
I have put your script in my code and I don't have the white screen now - my site shows up as expected but when I call record `<p>{record}</p>` in my html (in the same file) it shows `[Object Object]` instead of the record info.
![20231212_18h25m04s_grim](https://github.com/pocketbase/pocketbase/assets/118511626/ac313ee7-07ea-40ba-817e-3ae8196a28bb)
Sorry for the ugly site it is heavily a WIP. :)

**Reply by ganigeorgiev** [reply]
OK, this is not really a PocketBase issue.

> It shows [Object Object] instead of any info.

This is expected as `record` is a JS object and its string representation is `[object Object]`. It will be the same with any JS object.

In any case, you should be able to access the object fields using dot-notation, aka. `{record.id}`.
If you want to print the object as json, you can try `{JSON.stringify(record)}`.

While I want to help, please note that this is not a general dev support forum and JS or Svelte related questions are better suited for StackOverflow or in the related support channels of the tools you use.

**Reply by ItsTerm1n4l** [reply]
Ok I understand, Thank you very much for your help anyway. I didn't know enough about javascript to know where the fault was at so I apologise if used the wrong forum. 
Using `{JSON.stringify(record)}` gives me the record data I needed, Thank you.
Great work on pocketbase btw.
